crates/providers/blobstore-azblob/src/lib.rs

```diff
@@ -0,0 +1,529 @@
+        //
+        // TODO: az-blob-rust-sdk does not support continuation token
+        //
+
+        let client = match self.client(&ctx).await {
+            Ok(client) => client,
+            Err(e) => {
+                error!("failed to retrieve client: {e}");
+                return ListObjectsResponse {
+                    continuation: None,
+                    is_last: true,
+                    objects: Vec::new(),
+                };
+            }
+        };
+
+        let mut list_blob = client.container_client(req.container_id).list_blobs();
+
+        if let Some(max_items) = req.max_items {
+            if max_items > i32::MAX as u32 {
+                error!("max items too large");
+                return ListObjectsResponse {
+                    continuation: None,
+                    is_last: true,
+                    objects: vec![],
+                };
+            }
+            list_blob = list_blob.max_results(NonZeroU32::new(max_items).unwrap());
+        } else {
+            list_blob = list_blob.max_results(NonZeroU32::new(DEFAULT_MAX_ITEMS).unwrap());
+        }
+
+        let mut stream = list_blob.into_stream();
+
+        let mut all_blobs = Vec::new();
+
+        while let Some(response) = stream.next().await {
+            match response {
+                Ok(res) => {
+                    res.blobs.blobs().for_each(|c| {
+                        all_blobs.push(ObjectMetadata {
+                            container_id: c.name.clone(),
+                            last_modified: Some(Timestamp::from(c.properties.last_modified)),
+                            object_id: c.name.to_string(),
+                            content_length: c.properties.content_length,
+                            content_encoding: c.properties.content_encoding.clone(),
+                            content_type: Some(c.properties.content_type.to_string()),
+                        });
+                    });
+                }
+                Err(e) => {
+                    error!(error = %e, "failed to list containers");
+                }
+            };
+        }
+
+        ListObjectsResponse {
+            continuation: None,
+            is_last: true,
+            objects: all_blobs,
+        }
+    }
+
+    async fn remove_objects(
+        &self,
+        ctx: Context,
+        arg: RemoveObjectsRequest,
+    ) -> Vec<OperationResult> {
+        let client = match self.client(&ctx).await {
+            Ok(client) => client,
+            Err(e) => {
+                error!("failed to retrieve client: {e}");
+                return Vec::new();
+            }
+        };
+
+        let container_client = client.container_client(&arg.container_id);
+        let mut results = Vec::new();
+        for obj in arg.objects.iter() {
+            let blob_client = container_client.blob_client(obj);
+            match blob_client.delete().into_future().await {
```

Comment:
It doesn't seem like az-blob-rust-sdk provides a API for deleting multiple blobs in one req.